redash/query_runner/json_ds.py

```diff
@@ -71,6 +72,8 @@ def _apply_path_search(response, path):
         current_path = path_parts.pop()
         if current_path in response:
             response = response[current_path]
+        elif default is not None:
+            return default
```

Comment:
Won't this return on the first instance of `current_path not in response`? I am not sure having the return within the `while` loop is what we want, but rather we should rewrite this so that there's only one return statement in the function akin to `return response or default` , or something similar.